server/hotrod/src/main/java/org/infinispan/server/hotrod/ClientListenerRegistry.java

```diff
@@ -379,11 +372,11 @@ public void onCacheEvent(CacheEntryEvent<byte[], byte[]> event) {
             }
             Object k = event.getKey();
             Object v = event.getValue();
-            if (keyEncoder.isStorageFormatFilterable() && k != null) {
-               k = fromStorage(k, keyEncoder, keyWrapper);
+            if (keyDataConversion.isStorageFormatFilterable() && k != null) {
```

Comment:
is it necessary to test that k is not null ? the fromStorage already checks that it seems to me